monkey/monkey_island/cc/resources/propagation_credentials.py

```diff
@@ -1,15 +1,16 @@
+from flask import make_response
+
+from monkey_island.cc.repository import ICredentialsRepository
 from monkey_island.cc.resources.AbstractResource import AbstractResource
-from monkey_island.cc.services.config import ConfigService
 
 
 class PropagationCredentials(AbstractResource):
     urls = ["/api/propagation-credentials"]
 
-    def get(self):
-        config = ConfigService.get_flat_config(should_decrypt=True)
+    def __init__(self, credentials_repository: ICredentialsRepository):
+        self._credentials_repository = credentials_repository
 
-        propagation_credentials = ConfigService.get_config_propagation_credentials_from_flat_config(
-            config
-        )
+    def get(self):
+        propagation_credentials = self._credentials_repository.get_all_credentials()
 
-        return {"propagation_credentials": propagation_credentials}
+        return make_response({"propagation_credentials": propagation_credentials})
```

Comment:
Just return `propadation_credentials`. The endpoint is already called `/api/propagation-credentials`. The key that identifies the data is redundant. In other words, just return a list, not a dict.